hera_cal/io.py

```diff
@@ -537,6 +548,73 @@ def iterate_over_times(self, Nints=1, times=None):
             yield self.read(times=times[i:i + Nints])
 
 
+def load_flags(flagfile, filetype='h5', return_meta=False):
+    '''Load flags from a file and returns them as a DataContainer (for per-visibility flags)
+    or dictionary (for per-antenna or per-polarization flags). More than one spectral window
+    is not supported. Assumes times are evenly-spaced and in order for each baseline.
+    
+    Arguments:
+        flagfile: path to file containing flags and flagging metadata
+        filetype: either 'h5' or 'npz'. 'h5' assumes the file is readable as a hera_qm
+            UVFlag object in the 'flag' mode (could be by baseline, by antenna, or by 
+            polarization). 'npz' provides legacy support for the IDR2.1 flagging npzs,
+            but only for per-visibility flags.
+        return_meta: if True, return a metadata dictionary with, e.g., 'times', 'freqs', 'history'
+        
+    Returns:
+        flags: dictionary or DataContainer mapping keys to Ntimes x Nfreqs numpy arrays.
+            if 'h5' and 'baseline' mode or 'npz': DataContainer with keys like (0,1,'xx')
+            if 'h5' and 'antenna' mode: dictionary with keys like (0,'Jxx')
+            if 'h5' and 'waterfall' mode: dictionary with keys like 'Jxx'
+        meta: (only returned if return_meta is True)
+    '''
+    flags = {}
+    if filetype not in ['h5', 'npz']:
+        raise ValueError("filetype must be 'h5' or 'npz'.")
+    
+    elif filetype == 'h5':
+        from hera_qm import UVFlag
+        uvf = UVFlag(flagfile)
+        assert uvf.mode == 'flag', 'Must be in flag mode.'
```

Comment:
To help debugging, you may want this assert message to be a little more detailed.

Reply:
Fixed.